### DISCUSSION TITLE
What are some best practices for optimizing performance in Next.js applications?

### DISCUSSION THREAD

**Discussion by madhu22222** [vercel/next.js#71536] [Help] [answered]
### Summary

As a developer new to Next.js, I’m interested in understanding how to ensure that my applications perform efficiently. Are there specific techniques or strategies recommended for optimizing load times, improving SEO, and managing resources effectively? Any guidance on this topic would be greatly appreciated!

**Comment by madhavi-peddireddy** [comment, marked as answer]
To optimize performance in Next.js applications, consider implementing server-side rendering (SSR) or static site generation (SSG) for faster load times. Use the built-in image optimization features to serve images in the best format and size. Additionally, leverage dynamic imports for code splitting, minimize the use of large dependencies, and optimize API calls. Regularly analyze your application using tools like Lighthouse to identify bottlenecks and areas for improvement. Engaging with the community for tips and examples can also provide valuable insights.

**Comment by shecandevop** [comment]
@khuezy, what are your thoughts on best practices beside these, that are not so well-known to Next.js users?

**Reply by khuezy** [reply]
Sorry do we know each other?
Anyways, w/ the latest 15 features, on top of the SSR, I use dynamicIO and `use cache` and `cacheTag('thing')` to reduce db calls and improve load speeds when you have a page that loads multiple components and nested children.

**Reply by shecandevop** [reply]
@khuezy, thank you for your sharing your tips on optimizing Next.js . I've tagged you since your suggestions on Fly community have always been helpful.
